### DISCUSSION TITLE
Tif file with Alpha with JPEG Compression Issue !

### DISCUSSION THREAD

**Discussion by chukanet** [ImageMagick/ImageMagick#5913] [Digital Image Processing]
Hi, i'm facing an issue with ImageMagick.
I'm using windows and command (CMD) to call my conversion.

there is m'y command line: 
magick mogrify "C:\Users\myname\Desktop\tif\test\*.tif" -define tiff:alpha=unspecified -compress JPEG -quality 80 "C:\Users\myname\Desktop\tif\test\*.tif"

As you can see my goal is to Compress my tif cause i have some tif that make more than 250mb...
They containt Alpha channel that i don't wan't they where be flattened. I really need to keep alpha Channel :)

It's work but 25% of my image is black on the right after conversion done.
If i remove the alpha channel, i get my whole RGB file... but when i put alpha channel to the input file, i get a bad result.

tested on the lastest version..

regards
![issue](https://user-images.githubusercontent.com/25986103/210097584-f28b06e0-7480-4555-9b81-9f025d91ccc3.jpg)

**Comment by fmw42** [comment]
Please post your original input file. If necessary, zip it and post the zip file. If that does not work, then post to some free hosting service and put the URL here.

If you are using mogrify to process multiple input images, then there is no input for mogrify. It processes each image in the directory. So

`magick mogrify -define tiff:alpha=unspecified -compress JPEG -quality 80 "C:\Users\myname\Desktop\tif\test*.tif"`

If you need to write the output to a different directory then use the -path argument.

**Reply by chukanet** [reply]
hi thanks for the reply.
i'll try for the -define , but my exemple in top work for all the file in the desired folder :)

there is the file i've reduce it to 512*512 
but it's the same issue !

mogrify: Sum of Photometric type-related color channels and ExtraSamples doesn't match SamplesPerPixel. Defining non-color channels as ExtraSamples.. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/951.

[test.zip](https://github.com/ImageMagick/ImageMagick/files/10326102/test.zip)

**Reply by fmw42** [reply]
This command works fine for me on IM 7.1.0-56

`magick test.tif -define tiff:alpha=unspecified -compress JPEG -quality 80 x.tif`

[x.tif.zip](https://github.com/ImageMagick/ImageMagick/files/10326125/x.tif.zip)

What is your IM version?

Try my magick command. Perhaps mogrify, which is not updated for IM 7, is not handling the extra channel properly.

**Reply by fmw42** [reply]
This also works fine for me:

`magick mogrify -define tiff:alpha=unspecified -compress JPEG -quality 80 test.tif`

**Reply by chukanet** [reply]
the command work but not the result ... it's the same as me

**Reply by fmw42** [reply]
What is wrong?  I see no black on the right side even in Photoshop. Please clarify.  And what is your IM version?

**Reply by chukanet** [reply]
i've download your x.tif.zip and when i open it on photoshop it's get the black band on the right...
i've tested with ImageMagick-7.1.0-56 and ImageMagick-7.1.0-Q16

if i change compress JPEG to ZIP it's work well but  the file is so huge...


Ok i've tested with another soft and it's look OK... so strange maybe my Photoshop mistake

thx for your help you save my day !!

**Reply by fmw42** [reply]
I have opened all the files in Photoshop 2023 and they look fine. No black border on the right side. Check your Photoshop.
